test/k8sT/Policies.go

```diff
@@ -827,78 +736,157 @@ EOF`, k, v)
 		})
 	})
 
-	Context("KubernetesNetworkPolicy between server and client", func() {
+	Context("Namespaces policies", func() {
 
 		var (
-			policy = "allow-ns-b-via-namespace-selector"
+			err         error
+			ciliumPod   string
+			namespace   = "second"
+			appPods     map[string]string
+			appPodsNS   map[string]string
+			clusterIP   string
+			clusterIPNS string
+
+			demoPath         = helpers.ManifestGet("demo.yaml")
+			l3Policy         = helpers.ManifestGet("l3_l4_policy.yaml")
+			netpolNsSelector = fmt.Sprintf("%s -n %s", helpers.ManifestGet("netpol-namespace-selector.yaml"), namespace)
+			policy           = fmt.Sprintf("%s -n %s", l3Policy, namespace)
+			demoManifest     = fmt.Sprintf("%s -n %s", demoPath, namespace)
 		)
 
 		BeforeAll(func() {
-			By("Creating the namespace that will be used for the pods")
-			_, err := kubectl.CoreV1().Namespaces().Create(&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}})
-			Expect(err).NotTo(HaveOccurred())
-
-			By("Creating a simple server that serves on port 80 and 81.")
-			podServer, service = createServerPodAndService(kubectl, namespace, "server", []int{80, 81})
-
-			By("Waiting for pod ready", func() {
-				ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
-				defer cancel()
-				err := kubectl.WaitForPodReady(ctx, namespace, podServer.Name)
-				Expect(err).NotTo(HaveOccurred())
-			})
+
+			res := kubectl.NamespaceCreate(namespace)
+			res.ExpectSuccess("unable to create namespace %q", namespace)
+
+			res = kubectl.Exec(fmt.Sprintf("kubectl label namespaces/%[1]s nslabel=%[1]s", namespace))
+			res.ExpectSuccess("cannot create namespace labels")
+
+			res = kubectl.Apply(demoManifest)
+			res.ExpectSuccess("unable to apply manifest %s")
+
+			res = kubectl.Apply(demoPath)
+			res.ExpectSuccess("unable to apply manifest %s")
+
+			ciliumPod, err = kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+			Expect(err).Should(BeNil(), "cannot get CiliumPod")
+
+			err := kubectl.WaitforPods(namespace, "-l zgroup=testapp", 300)
+			Expect(err).To(BeNil(), "testapp pods are not ready after timeout")
+
+			err = kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", 300)
+			Expect(err).To(BeNil(), "testapp pods are not ready after timeout")
```

Comment:
Since we wait for testapp pods to be ready in separate namespaces, include the namespace in the error message.